libmscore/textbase.cpp

```diff
@@ -2609,6 +2612,59 @@ TextCursor* TextBase::cursor(const EditData& ed)
       return &ted->cursor;
       }
 
+//---------------------------------------------------------
+//   applyRotation
+//---------------------------------------------------------
+
+void TextBase::applyRotation()
+      {
+      //
+      // because the bounding box is rotated by resizing the rectangle and the QPainter is rotated normally (TextBase::draw(Qpainter*)),
+      // there is small offset between the text and the bounding box of the text, that's where offsetCorrection comes in
+      //
+      qreal offsetCorrection = (_basebbox.height() / 2) * abs(sin(qDegreesToRadians(static_cast<double>(_rotation))));
```

Comment:
I've read somewhere that the "correct" way of applying a rotation was to first apply a translation to the center, then rotate and translate back to the original point.
Hope this can be simplified with it.